tests/nightly/gpu/test_t5.py

```diff
@@ -0,0 +1,270 @@
+        )
+        self.assertEqual(translation, expected_translation)
+
+    @set_device
+    def test_translation_en_to_fr(self):
+        """
+        From Huggingface.
+        """
+        cfg_name = 'translation_en_to_fr'
+        cfg = TASK_CONFIGS[cfg_name]
+        self.agent.opt['t5_generation_config'] = cfg_name
+
+        en_text = ' This image section from an infrared recording by the Spitzer telescope shows a "family portrait" of countless generations of stars: the oldest stars are seen as blue dots. '
+
+        text_vec = (
+            torch.LongTensor(self.agent.dict.txt2vec(cfg['prefix'] + en_text))
+            .unsqueeze(0)
+            .to(device)
+        )
+
+        generations, _ = self.agent._generate(
+            Batch(text_vec=text_vec),
+            beam_size=4,
+            max_ts=100,
+            overrides={
+                'max_length': 100,
+                'no_repeat_ngram_size': 3,
+                'length_penalty': 2.0,
+            },
+        )
+
+        translation = self.agent.dict.vec2txt(
+            generations[0][0], clean_up_tokenization_spaces=False
+        )
+        new_truncated_translation = (
+            "Cette section d'images provenant de l'enregistrement infrarouge effectué par le télescope Spitzer montre "
+            "un "
+            "« portrait familial » de générations innombrables d’étoiles : les plus anciennes sont observées "
+            "sous forme "
+            "de points bleus."
+        )
+
+        self.assertEqual(translation, new_truncated_translation)
+
+    @set_device
+    def test_translation_en_to_ro(self):
+        """
+        From Huggingface.
+        """
+        cfg_name = 'translation_en_to_ro'
+        cfg = TASK_CONFIGS[cfg_name]
+        self.agent.opt['t5_generation_config'] = cfg_name
+        en_text = "Taco Bell said it plans to add 2,000 locations in the US by 2022."
+        expected_translation = "Taco Bell a declarat că intenţionează să adauge 2 000 de locaţii în SUA până în 2022."
+
+        text_vec = (
+            torch.LongTensor(self.agent.dict.txt2vec(cfg['prefix'] + en_text))
+            .unsqueeze(0)
+            .to(device)
+        )
+        generations, _ = self.agent._generate(
+            Batch(text_vec=text_vec), beam_size=4, max_ts=300
+        )
+        translation = self.agent.dict.vec2txt(
+            generations[0][0], clean_up_tokenization_spaces=False
+        )
+        self.assertEqual(translation, expected_translation)
+
+    @set_device
+    def test_t5_gen(self):
+        """
+        Test out-of-the-box T5 generation.
+        """
+        opt = ParlaiParser(True, True).parse_args(['--model', 't5'])
+        t5 = create_agent(opt)
+        text = "Don't have a cow, Man!"
+        obs = {"text": text, 'episode_done': True}
+        t5.observe(obs)
+        act = t5.act()
+
+        self.assertEqual(act['text'], text)
+
+    @set_device
+    @testing_utils.retry(ntries=3, log_retry=True)
```

Comment:
I'd prefer if we didn't. Could we do one of overfit datasets instead?

Reply:
yeah